skrub/_join_utils.py

```diff
@@ -68,3 +85,61 @@ def check_missing_columns(table, key, table_name):
         "The following columns cannot be used for joining because they do not exist"
         f" in {table_name}:\n{missing_columns}"
     )
+
+
+def check_column_name_duplicates(main_columns, aux_columns, suffix, table_names={}):
```

Comment:
2: oops good catch I added it to start addressing this [comment](https://github.com/skrub-data/skrub/pull/821#discussion_r1404345563) but then I forgot to actually use it :sweat_smile:

Reply:
>  or maybe a special value for 'suffix' that means 'figure out a suffix that makes the column unique'

Yes, this is a good solution. A longish random string might do the trick! (or maybe a date, as it would allow debugging?)